### PR TITLE
Add JerryScript 2.3.0

### PR DESCRIPTION
**Pull request by ossy-szeged**

### CODE REVIEW

File: data-es6.js [hunk 1]
```diff
@@ -21325,6 +21480,7 @@ exports.tests = [
         graalvm20: true,
         graalvm20_1: true,
         jerryscript2_0: true,
+        jerryscript2_3_0: true,
```
Comment (ljharb) [inline data-es6.js]: this shouldn't be needed, since it should be inherited from 2.0

File: data-es6.js [hunk 2]
```diff
@@ -21598,6 +21764,7 @@ exports.tests = [
...
```
Comment (ljharb) [inline data-es6.js]: also here

File: environments.json [hunk 3]
```diff
@@ -3694,12 +3694,25 @@
...
+    "short": "JrS 2.3.0",
+    "platformtype": "engine",
+    "release": "2020-06-12",
+    "obsolete": false,
+    "test_suites": [
+      "es5",
+      "es6",
+      "es2016plus"
+    ]
+  },
```
Comment (ljharb) [inline environments.json:3715]:
```suggestion
  "jerryscript2_3_0": {
    "full": "JerryScript 2.3.0",
    "family": "JerryScript",
    "short": "JrS 2.3.0",
    "platformtype": "engine",
    "release": "2020-06-12",
    "equals": "jerryscript2_2_0",
    "obsolete": false,
    "test_suites": [
      "es5",
      "es6",
      "es2016plus"
    ]
  },
```

File: environments.json [hunk 4]
```diff
@@ -3694,12 +3694,25 @@
     "short": "JrS 2.2.0",
     "platformtype": "engine",
     "release": "2020-02-11",
-    "obsolete": false,
+    "obsolete": true,
```
Comment (ljharb) [inline environments.json:3697]:
```suggestion
    "obsolete": true,
    "equals": "jerryscript2_1_0",
```

**Review by ljharb — CHANGES_REQUESTED** [review]
Please also add the "equals" for all of the jerryscripts so the chain is complete :-)

File: data-es6.js (hunk 1 above)
Comment (ossy-szeged) [reply data-es6.js]: Good catch, it is inherited properly, it is only a mistake. I removed this change.

File: data-es6.js (hunk 2 above)
Comment (ossy-szeged) [reply data-es6.js]: also removed

File: environments.json (hunk 3 above)
Comment (ossy-szeged) [reply environments.json:3715]: It is inherited properly without this change.

File: environments.json (hunk 4 above)
Comment (ossy-szeged) [reply environments.json:3697]: it isn't needed too

File: jerryscript.js [hunk 5]
```diff
@@ -79,19 +79,43 @@ var asyncTestHelperHead =
...
+'  job.cb(job.cbarg)\n' +
+'}\n' +
+'\n' +
+'const catchCb = job => {\n' +
+'  jobRunner(job);\n' +
+'}\n' +
+'\n' +
+'function jobRunner(job){\n' +
+'  return new Promise((resolve, reject) => {\n' +
+'    let diff = new Date().getTime() - job.startTime;\n' +
```
Comment (ljharb) [inline jerryscript.js]:
```suggestion
'    let diff = Date.now() - job.startTime;\n' +
```

**Review by ljharb — APPROVED** [review]

File: jerryscript.js [hunk 6]
```diff
@@ -79,19 +79,43 @@ var asyncTestHelperHead =
 '  asyncPassed = true;\n' +
 '}\n' +
 '\n' +
-'function setTimeout(cb, time, ...cbarg) {\n' +
+'function setTimeout(cb, time, cbarg) {\n' +
 '  if (!jobqueue[time]) {\n' +
 '    jobqueue[time] = [];\n' +
 '  }\n' +
-'  jobqueue[time].push({cb, cbarg});\n' +
+'  jobqueue[time].push({cb, cbarg, startTime: new Date().getTime(), timeout: time});\n' +
```
Comment (ljharb) [inline jerryscript.js]:
```suggestion
'  jobqueue[time].push({cb, cbarg, startTime: Date.now(), timeout: time});\n' +
```
sorry, missed this one

**Review by chicoxyzzy — APPROVED** [review]

**Pull request merged**